### PR TITLE
Story/quad15

### PR DESCRIPTION
**Pull request by gunpreetsingh**
Creating a pull request as per review comments.

### CODE REVIEW

File: Quadriga/src/main/java/edu/asu/spring/quadriga/web/workspace/ArchiveWSController.java [hunk 1]
```diff
@@ -51,106 +53,27 @@ protected void initBinder(WebDataBinder binder)
...
+	 * @throws QuadrigaAccessException
+	 * @author Kiran Kumar Batna
 	 */
-	@AccessPolicies({ @ElementAccessPolicy(type = CheckedElementType.PROJECT,paramIndex = 3, userRole = {RoleNames.ROLE_COLLABORATOR_ADMIN,RoleNames.ROLE_PROJ_COLLABORATOR_ADMIN,RoleNames.ROLE_PROJ_COLLABORATOR_CONTRIBUTOR})
-	,@ElementAccessPolicy(type=CheckedElementType.WORKSPACE,paramIndex=0,userRole={})})
-	@RequestMapping(value = "auth/workbench/{projectid}/archiveworkspace", method = RequestMethod.POST)
-	public ModelAndView archiveWorkspace(@Validated @ModelAttribute("workspaceform") ModifyWorkspaceForm workspaceForm,BindingResult result,
-			@PathVariable("projectid") String projectid,Principal principal) throws QuadrigaStorageException, QuadrigaAccessException
+	@AccessPolicies({ @ElementAccessPolicy(type = CheckedElementType.PROJECT,paramIndex = 1, userRole = {RoleNames.ROLE_COLLABORATOR_ADMIN, RoleNames.ROLE_PROJ_COLLABORATOR_ADMIN, RoleNames.ROLE_PROJ_COLLABORATOR_CONTRIBUTOR})
+	,@ElementAccessPolicy(type=CheckedElementType.WORKSPACE,paramIndex=2,userRole={})})
```
Comment (jdamerow) [inline Quadriga/src/main/java/edu/asu/spring/quadriga/web/workspace/ArchiveWSController.java:68]: There are still the roles missing.

File: Quadriga/src/main/java/edu/asu/spring/quadriga/web/workspace/ArchiveWSController.java [hunk 2]
```diff
@@ -51,106 +53,27 @@ protected void initBinder(WebDataBinder binder)
...
-			model.getModelMap().put("wsprojectid", projectid);	
-		}
-
-      return model; 
-
+		ModelAndView model = new ModelAndView("redirect:/auth/workbench/"+ projectid);
+		archiveWSManager.archiveWorkspace(workspaceid, principal.getName());		
+		redirectAttributes.addFlashAttribute("show_success_alert", true);
+		redirectAttributes.addFlashAttribute("success_alert_msg","The workspace has been successfully archived.");
+        return model;
```
Comment (jdamerow) [inline Quadriga/src/main/java/edu/asu/spring/quadriga/web/workspace/ArchiveWSController.java:76]: Why is that not indented?

File: Quadriga/src/main/java/edu/asu/spring/quadriga/service/impl/workspace/ListWSManager.java [hunk 3]
```diff
@@ -324,5 +323,12 @@ public boolean getDeactiveStatus(String workspaceId) throws QuadrigaStorageExcep
...
     }
+    @Transactional
+    public boolean getArchiveStatus(String workspaceId) throws QuadrigaStorageException {
+        WorkspaceDTO wsDto = workspaceDao.getWorkspaceDTO(workspaceId.trim());
+        if (wsDto != null)        
+            return wsDto.getIsarchived();        
+        return false;
+    }
 
 }
```
Comment (jdamerow) [inline Quadriga/src/main/java/edu/asu/spring/quadriga/service/impl/workspace/ListWSManager.java:334]: Let's write a test class for this class.

**Comment by jdamerow** [comment]
see comments

File: Quadriga/src/main/java/edu/asu/spring/quadriga/web/workspace/ArchiveWSController.java (hunk 1 above)
Comment (gunpreetsingh) [inline Quadriga/src/main/java/edu/asu/spring/quadriga/web/workspace/ArchiveWSController.java:68]: roles added

File: Quadriga/src/main/java/edu/asu/spring/quadriga/web/workspace/ArchiveWSController.java (hunk 2 above)
Comment (gunpreetsingh) [inline Quadriga/src/main/java/edu/asu/spring/quadriga/web/workspace/ArchiveWSController.java:76]: Code is indented now.

File: Quadriga/src/main/java/edu/asu/spring/quadriga/service/impl/workspace/ListWSManager.java (hunk 3 above)
Comment (gunpreetsingh) [inline Quadriga/src/main/java/edu/asu/spring/quadriga/service/impl/workspace/ListWSManager.java:334]: Done

**Pull request closed without merging**